server/server/src/sync_on_central/mod.rs

```diff
@@ -103,68 +101,33 @@ async fn download_file(
 
 #[put("/sync/upload_file/{file_id}")]
 async fn upload_file(
-    payload: Multipart,
+    MultipartForm(form): MultipartForm<SyncUploadFileRequestV6>,
     settings: Data<Settings>,
     service_provider: Data<ServiceProvider>,
     path: web::Path<String>,
 ) -> Result<HttpResponse, Error> {
-    let db_connection = service_provider
-        .connection()
-        .map_err(|err| InternalError::new(err, StatusCode::INTERNAL_SERVER_ERROR))?;
-
     let file_id = path.into_inner();
-
-    let repo = SyncFileReferenceRowRepository::new(&db_connection);
     log::info!("Receiving a file via sync : {}", file_id);
-    let mut sync_file_reference = repo
-        .find_one_by_id(&file_id)
-        .map_err(|err| InternalError::new(err, StatusCode::INTERNAL_SERVER_ERROR))?
-        .ok_or({
-            log::error!(
-                "Sync File Reference not found, can't upload until this is synced: {}",
-                file_id
-            );
-            InternalError::new(
-                "Sync File Reference not found, can't upload until this is synced",
-                StatusCode::NOT_FOUND,
-            )
-        })?;
-
-    let files = handle_file_upload(
-        payload,
-        settings,
-        StaticFileCategory::SyncFile(
-            sync_file_reference.table_name.clone(),
-            sync_file_reference.record_id.clone(),
-        ),
-        Some(file_id),
-    )
-    .await?;
-
-    let repo = SyncFileReferenceRowRepository::new(&db_connection);
-    if files.len() != 1 {
-        log::error!(
-            "Incorrect sync file upload received: Expected to see 1 file uploaded, but got {}",
-            files.len()
-        );
-    }
-
-    for file in files.clone() {
-        sync_file_reference.uploaded_bytes += file.bytes;
-        let result = repo.upsert_one(&sync_file_reference);
-        match result {
-            Ok(_) => {}
-            Err(err) => {
-                log::error!(
-                    "Error saving sync file reference after sync upload: {}",
-                    err
-                );
 
-                return Err(InternalError::new(err, StatusCode::INTERNAL_SERVER_ERROR).into());
+    let result = sync_on_central::upload_file(&settings, &service_provider, file_id, form).await;
+    match result {
```

Comment:
I think here we can follow the existing pattern of push and pull response ? Otherwise we would have three patterns for doing it, pull/push, download-file (I didn't know how to do multipart response, if that's even a thing, so the error is serialized in an error response), and then this one with statuses.

Reply:
Do you mean I should have a return type like `SyncUploadSuccessV6`?
Not entirely sure what you mean sorry!